test/boost/README.md

```diff
@@ -0,0 +1,80 @@
+# Scylla unit tests using C++ and the Boost test framework
+
+The source files in this directory are Scylla unit tests written in C++
+using the Boost.Test framework. These unit tests come in three flavors:
+
+1. Some simple tests that check stand-alone C++ functions or classes use
+   Boost's `BOOST_AUTO_TEST_CASE`.
+
+2. Some tests require Seastar features, and need to be declared with Seastar's
+   extensions to Boost.Test, namely `SEASTAR_TEST_CASE`.
+
+3. Even more elaborate tests require not just a functioning Seastar environment
+   but also a complete (or partial) Scylla environment. Those tests use the
+   `do_with_cql_env()` or `do_with_cql_env_thread()` function to set up a
+   mostly-functioning environment behaving like a single-node Scylla, in
+   which the test can run.
+
+While we have many tests of the third flavor, writing new tests of this
+type should be reserved to *white box* tests - tests where it is necessary
+to inspect or control Scylla internals that do not have user-facing APIs
+such as CQL. In contrast, black-box tests - tests that **can** be written
+only using user-facing APIs, should be written in one of newer test frameworks
+that we offer - such as test/cql-pytest or test/alternator (in Python, using
+the CQL or DynamoDB APIs respectively) or test/cql (using textual CQL
+commands), or - if more than one Scylla node is needed for a test - using
+the test/topology* framework.
+
+# Running tests
+
+Because these are C++ tests, they need to be compiled before running.
+To compile a single test executable `row_cache_test`, use a command like
+```
+ninja build/dev/test/boost/row_cache_test
+```
+You can also use `ninja dev-test` to build _all_ C++ tests, or use
+`ninja deb-build` to build the C++ tests and also the full Scylla executable
+(however, note that full Scylla executable isn't needed to run Boost tests).
+
+Replace "dev" by "debug" or "release" in the examples above and below to
+use the "debug" build mode (which, importantly, compiles the test with
+ASAN and UBSAN enabling on and helps catch difficult-to-catch use-after-free
+bugs) or the "release" build mode (optimized for run speed).
+
+To run an entire test file `row_cache_test`, including all its test
+functions, use a command like:
+```
+build/dev/test/boost/row_cache_test -- -c1 -m1G 
+```
+
+to run a single test function `test_reproduce_18045()` from the longer test
```

Comment:
I think some of this is repeating the contents of docs/dev/testing.md, I don't know if this is intended or not. one of way of handling the redundancy would be to reference docs/dev/testing.md

Reply:
docs/dev/testing.md focuses on test.py, and doesn't have any of the above examples of how to run a test directly without test.py, nor how to build the C++ tests. It also doesn't even mention the fact we have three "types" of Boost tests (not using Seastar, using Seastar, and using CQL) which is important to know. So what duplication bothered you exactly?

And in general, I think every directory needs its own README.md to explain to new developers what these files are, how to work with them and how to use them. I don't think this explanation is at all redundant or unnecessary, or that having a README.md saying "This is one of the test directories, take a look at docs/dev/testing.md for an explanation of all test directories" would have been enough.